rust/worker/src/execution/operators/fetch_segment.rs

```diff
@@ -0,0 +1,194 @@
+use chroma_blockstore::provider::BlockfileProvider;
+use chroma_error::{ChromaError, ErrorCodes};
+use chroma_index::{hnsw_provider::HnswIndexProvider, IndexConfig, IndexConfigFromSegmentError};
+use chroma_types::{Collection, Segment, SegmentScope, SegmentType};
+use thiserror::Error;
+use tonic::async_trait;
+use tracing::trace;
+use uuid::Uuid;
+
+use crate::{
+    execution::operator::Operator,
+    segment::{
+        distributed_hnsw_segment::{
+            DistributedHNSWSegmentFromSegmentError, DistributedHNSWSegmentReader,
+        },
+        metadata_segment::{MetadataSegmentError, MetadataSegmentReader},
+        record_segment::{RecordSegmentReader, RecordSegmentReaderCreationError},
+    },
+    sysdb::sysdb::{GetCollectionsError, GetSegmentsError, SysDb},
+};
+
+#[derive(Clone, Debug)]
+pub struct FetchSegmentOperator {
```

Comment:
Can we tag this operator type as IO since it does I/O so that it does not run on worker threads? See get_type() in operator.rs for an example

Reply:
Yes, it should be I/O

Reply:
I've updated the operator type.